internal/service/influxdb/resource_influxdb_database.go

```diff
@@ -0,0 +1,172 @@
+package influxdb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aiven/aiven-go-client"
+	"github.com/aiven/terraform-provider-aiven/internal/schemautil"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var aivenInfluxDBDatabaseSchema = map[string]*schema.Schema{
+	"project":      schemautil.CommonSchemaProjectReference,
+	"service_name": schemautil.CommonSchemaServiceNameReference,
+	"database_name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: schemautil.Complex("The name of the service database.").ForceNew().Build(),
+	},
+	"termination_protection": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: schemautil.Complex(`It is a Terraform client-side deletion protections, which prevents the database from being deleted by Terraform. It is recommended to enable this for any production databases containing critical data.`).DefaultValue(false).Build(),
+	},
+}
+
+func ResourceInfluxDBDatabase() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The InfluxDB Database resource allows the creation and management of Aiven InfluxDB Databases.",
+		CreateContext: resourceDatabaseCreate,
+		ReadContext:   resourceDatabaseRead,
+		DeleteContext: resourceDatabaseDelete,
+		UpdateContext: resourceDatabaseUpdate,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceDatabaseState,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Delete: schema.DefaultTimeout(2 * time.Minute),
+		},
+
+		Schema: aivenInfluxDBDatabaseSchema,
+	}
+}
+
+func resourceDatabaseCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
```

Comment:
Perhaps this can be named something like `resourceInfluxDBCreate`, to make it more specific. And the same for other functions (Read, Delete, Update....) and similarly for PG and MySQL